crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -283,6 +283,27 @@ impl GitCommandProvider {
 
         Ok(OsString::from_vec(out.stdout))
     }
+
+    // checks if the specified revision identifies a commit in the repo
+    pub async fn contains_commit(&self, rev: &str) -> Result<bool, GitCommandError> {
+        let result = GitCommandProvider::run_command(
+            GitCommandProvider::new_command(&self.workdir)
+                .arg("rev-parse")
+                .arg("--quiet")
+                .arg("--verify")
+                .arg(format!("{}^{{commit}}", rev)),
+        )
+        .await;
+        match result {
+            Ok(_) => Ok(true),
+            Err(GitCommandError::BadExit(_, stdout, stderr))
+                if stdout.is_empty() && stderr.is_empty() =>
+            {
+                Ok(false)
+            },
+            Err(e) => Err(e),
+        }
```

Comment:
@ysndr given we don't want everything to be generic should I just start implementing methods on `GitCommandProvider` like this?